inc/Engine/Media/Fonts/Frontend/Controller.php

```diff
@@ -207,6 +215,42 @@ public function disable_google_fonts_preload( $disable ): bool {
 		return true;
 	}
 
+	/**
+	 * Get the list of patterns to exclude from media fonts rewrite.
+	 *
+	 * @return string[]
+	 */
+	private function get_exclusions(): array {
+		/**
+		 * Filters the list of patterns to exclude from media font rewrite.
+		 *
+		 * @since 3.18
+		 *
+		 * @param string[] $exclusions The list of patterns to exclude from media fonts.
+		 */
+		return wpm_apply_filters_typed( 'string[]', 'rocket_exclude_locally_host_fonts', [] );
+	}
+
+	/**
+	 * Checks if a URL is excluded based on the provided exclusions.
+	 *
+	 * @param string   $url        The URL to check.
+	 * @param string[] $exclusions The list of exclusions.
+	 *
+	 * @return bool True if the URL is excluded, false otherwise.
+	 */
+	private function is_excluded( string $url, array $exclusions ): bool {
+		if ( ! empty( $exclusions ) ) {
+			foreach ( $exclusions as $exclusion ) {
+				if ( preg_match( '/' . $exclusion . '/', $url ) ) {
```

Comment:
I don't know if one solution is better than the other, but to avoid the loop we previously did the following:
`$exclusions_str = implode( '|', $exclusions );`